### ISSUE TITLE
It would be great is versioning:classify could also read/set release.version

### ISSUE THREAD

**Issue by talios** [sviperll/ozymandias#2]
Hi, just looking at this plugin - looks great.

One thing, it would be great if `versioning:prompt` could provide some form of automatic increment, so if my current version was `1.34-alpha1-SNAPSHOT` the prompt would give me `1.34-alpha2` by default.

I see in the code there is something for discovering version numbers, so maybe I'm doing something wrong in my setup?

Currently the plugin seems to default to `final` all the time, rather than the current `release.kind`.

**Comment by sviperll** [comment]
My current usage is that I have version like 1.34-alpha1-successor-SNAPSHOT or 1.34-alpha1-next-SNAPSHOT as a current version. And with _that_ you get automatic increment. After a release I set current version to `<released version>-successor-SNAPSHOT`.

When you are determined to release particular version number, before actual release you can change current version from `<last released version>-successor-SNAPSHOT` to `<next release version>-SNAPSHOT` and `versioning:promt` will give you `<next release version>` when you ask.

As for defaulting to final, I've tried different defaults: defaulting to `beta` when current kind is `alpha`, defaulting to current kind, etc. This all seems too arbitrary. May be it's better to have no default...

**Comment by sviperll** [comment]
@talios, have you managed to set up versioning-plugin in alignment to your needs? You may provide some information about your use case, so that I will be able to include required documentation into documentation.

Besides you can look into adt4j's [pom.xml](https://github.com/sviperll/adt4j/blob/master/pom.xml) for details about versioning-plugin set-up in conjunction with release-plugin.

**Comment by talios** [comment]
@sviperll My bad - been snowed under at work with other things that I've not had a chance to revisit this.

Unfortunately I think ${manager} wants to use different naming to what the plugin users, like CR/GA etc.

**Comment by sviperll** [comment]
Hmm, I can probably add naming customization for this plugin. It will be great if you can outline your particular naming scheme, so that I can come up with customization scheme that will allow to use both current and your required naming.

**Comment by sviperll** [comment]
I've released version 0.22 of ozymandias suit which includes versioning plugin with customizable versioning schema.

You can play with versioning now from command line like shown below. `versioning.order` parameter defines preferred naming alone with the version order up to final release version.

There is a preconfigured set of aliases for possible version suffixes, but it can be extended with plugin configuration. See [VersioningMojo.java](https://github.com/sviperll/ozymandias/blob/master/versioning-maven-plugin/src/main/java/com/github/sviperll/maven/plugin/versioning/VersioningMojo.java) file for example configuration. I'm going to write some better configuration, but I haven't got the time, yet.

```
$ mvn -Dversioning.order='Alpha,Beta,CR,GA' -Dversioning.version=2.0.beta.1.0-next-SNAPSHOT com.github.sviperll:versioning-maven-plugin:0.22:prompt
[INFO] Current version is 2.0.beta.1.0-next-SNAPSHOT
Select version kind: Alpha, Beta, CR (release candidate), GA (Alpha/a/alpha/Beta/b/beta/CR/RC/rc/Final/GA/final) GA: : rc
Choose variant
--------------

1. 2.0.CR
2. 2.0.CR1
3. <custom>

$ mvn -Dversioning.order='Alpha,Beta,CR,GA' -Dversioning.version=2.0.beta.1.0-next-SNAPSHOT com.github.sviperll:versioning-maven-plugin:0.22:prompt
[INFO] Current version is 2.0.beta.1.0-next-SNAPSHOT
Select version kind: Alpha, Beta, CR (release candidate), GA (Alpha/a/alpha/Beta/b/beta/CR/RC/rc/Final/GA/final) GA: : beta
Choose variant
--------------

1. 2.0.beta.2
2. 2.0.beta.2.0
3. 2.0.beta.1.1
4. <custom>

```

**Comment by sviperll** [comment]
I've added some documentation about new version customization capabilities.
